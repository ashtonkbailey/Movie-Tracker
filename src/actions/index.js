// movie actions
/////// add movies
export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies
})

// user actions
//////// add favorite, delete favorite, sign in, sign out, create user

// export const addUser = (userObject) => ({
//   type: 'ADD_USER',
//   user
// })

// export const logInUser = (userObject) => ({
//   type: 'LOG_IN_USER',
//   user
// })