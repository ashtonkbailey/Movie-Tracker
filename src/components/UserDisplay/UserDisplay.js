import React from 'react';
import '../../index.scss';
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

const UserDisplay = (props) => {
  const { user } = props
  let button

  if (user.name) {
    button = <button>Log Out</button>
  } else {
    button = 
      <div>
        <Link to='/signup'>
          <button>Sign Up</button>
        </Link>
        <Link to='/login'>
          <button>Log In</button>
        </Link>
      </div>
  }

  return(
    <div className='user-display'>
      {button}
    </div>
  )
}

const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps, null)(UserDisplay);
