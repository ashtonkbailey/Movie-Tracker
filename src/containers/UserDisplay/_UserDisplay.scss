.user-display {
  position: absolute;
  top: 10px;
  right: 10px;

}
.control-btns {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.control {
  font-family: 'Sanchez', serif;
  font-size: 1rem;
  margin: 0 auto 10px;
  width: 120px;
  height: 32px;
  background: #950202;
  border: none;
  border-radius: 2px;
  color: #fff;
  transition: .4s ease;
  cursor: pointer;
  outline: none;
  box-shadow: 0 0 6px 1px rgba(0, 0, 0, 0.5),
        inset 0 0 6px 1px rgba(0, 0, 0, 0.4);

  &:hover {
    color: rgb(255, 216, 58);
    font-size: 1.05rem;
    text-shadow: 0 0 3px rgba(0, 0, 0, 0.4);
  }
}