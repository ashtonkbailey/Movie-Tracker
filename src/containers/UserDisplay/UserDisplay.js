import React from 'react';
import '../../index.scss';
import { connect } from 'react-redux'
import { Link } from 'react-router-dom';
import { logOutUser } from '../../actions/index';
import PropTypes from 'prop-types';

const UserDisplay = (props) => {
  const { user } = props
  let button

  if (user.name) {
    button = 
      <button 
        className='control'
        onClick={props.logOutUser}
      >Log Out</button>
  } else {
    button = 
      <div className='control-btns'>
        <Link to='/signup'>
          <button className='control'>Sign Up</button>
        </Link>
        <Link to='/login'>
          <button className='control'>Log In</button>
        </Link>
      </div>
  }

  return(
    <div className='user-display'>
      {button}
    </div>
  )
}

UserDisplay.propTypes = {
  user: PropTypes.object.isRequired,
  logOutUser: PropTypes.func.isRequired
}

const mapStateToProps = (state) => ({
  user: state.user
})

const mapDispatchToProps = (dispatch) => ({
  logOutUser: () => dispatch(logOutUser())
})

export default connect(mapStateToProps, mapDispatchToProps)(UserDisplay);
