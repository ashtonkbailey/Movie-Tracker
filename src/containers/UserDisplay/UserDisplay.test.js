import React from 'react';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps, UserDisplay } from './UserDisplay'
import { logOutUser, clearFavorites } from '../../actions/index';

describe('UserDisplay', () => {
  let mockUser

  beforeEach(() => {
    mockUser = { name: 'Bob', email: 'bob@gmail.com', password: 'bob' }
  })

  describe('UserDisplay Component', () => {
    it('should match the snapshot', () => {
      const mockLogOutUser = jest.fn()
      const mockClearFavorites = jest.fn()
      const wrapper = shallow(<UserDisplay user={mockUser} logOutUser={mockLogOutUser} clearFavorites={mockClearFavorites}/>)
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with no user', () => {
      const noUser = { name: '', email: '', password: '' }
      const mockLogOutUser = jest.fn()
      const mockClearFavorites = jest.fn()
      const wrapper = shallow(<UserDisplay user={noUser} logOutUser={mockLogOutUser} clearFavorites={mockClearFavorites}/>)
      expect(wrapper).toMatchSnapshot()
    })

    it('should call logOutUser and clearFavorites', () => {
      const mockLogOutUser = jest.fn()
      const mockClearFavorites = jest.fn()
      const wrapper = shallow(<UserDisplay user={mockUser} logOutUser={mockLogOutUser} clearFavorites={mockClearFavorites}/>)
      wrapper.find('Link').first().simulate('click')
      expect(mockLogOutUser).toBeCalled()
      expect(mockClearFavorites).toBeCalled()
    })
  })

  describe('mapStateToProps', () => {
    it('should return a props object with data from store', () => {
      const mockState = { user: mockUser }
      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(mockState)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should return a props object with a method logOutUser', () => {
      const mockDispatch = jest.fn()
      const expected = logOutUser()
      const expected2 = clearFavorites()

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.logOutUser()
      mappedProps.clearFavorites()

      expect(mockDispatch).toHaveBeenNthCalledWith(1, expected)
      expect(mockDispatch).toHaveBeenNthCalledWith(2, expected2)
    })
  })
})