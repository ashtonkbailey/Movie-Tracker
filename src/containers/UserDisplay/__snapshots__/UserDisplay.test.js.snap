// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDisplay UserDisplay Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDisplay
    clearFavorites={[MockFunction]}
    logOutUser={[MockFunction]}
    user={
      Object {
        "email": "bob@gmail.com",
        "name": "Bob",
        "password": "bob",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        onClick={[Function]}
        replace={false}
        to="/"
      >
        <button
          className="control"
        >
          Log Out
        </button>
      </Link>,
      "className": "user-display",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <button
          className="control"
        >
          Log Out
        </button>,
        "onClick": [Function],
        "replace": false,
        "to": "/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Log Out",
          "className": "control",
        },
        "ref": null,
        "rendered": "Log Out",
        "type": "button",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          onClick={[Function]}
          replace={false}
          to="/"
        >
          <button
            className="control"
          >
            Log Out
          </button>
        </Link>,
        "className": "user-display",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="control"
          >
            Log Out
          </button>,
          "onClick": [Function],
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log Out",
            "className": "control",
          },
          "ref": null,
          "rendered": "Log Out",
          "type": "button",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`UserDisplay UserDisplay Component should match the snapshot with no user 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDisplay
    clearFavorites={[MockFunction]}
    logOutUser={[MockFunction]}
    user={
      Object {
        "email": "",
        "name": "",
        "password": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="control-btns"
      >
        <Link
          replace={false}
          to="/signup"
        >
          <button
            className="control"
          >
            Sign Up
          </button>
        </Link>
        <Link
          replace={false}
          to="/login"
        >
          <button
            className="control"
          >
            Log In
          </button>
        </Link>
      </div>,
      "className": "user-display",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/signup"
          >
            <button
              className="control"
            >
              Sign Up
            </button>
          </Link>,
          <Link
            replace={false}
            to="/login"
          >
            <button
              className="control"
            >
              Log In
            </button>
          </Link>,
        ],
        "className": "control-btns",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="control"
            >
              Sign Up
            </button>,
            "replace": false,
            "to": "/signup",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "className": "control",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="control"
            >
              Log In
            </button>,
            "replace": false,
            "to": "/login",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log In",
              "className": "control",
            },
            "ref": null,
            "rendered": "Log In",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="control-btns"
        >
          <Link
            replace={false}
            to="/signup"
          >
            <button
              className="control"
            >
              Sign Up
            </button>
          </Link>
          <Link
            replace={false}
            to="/login"
          >
            <button
              className="control"
            >
              Log In
            </button>
          </Link>
        </div>,
        "className": "user-display",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/signup"
            >
              <button
                className="control"
              >
                Sign Up
              </button>
            </Link>,
            <Link
              replace={false}
              to="/login"
            >
              <button
                className="control"
              >
                Log In
              </button>
            </Link>,
          ],
          "className": "control-btns",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="control"
              >
                Sign Up
              </button>,
              "replace": false,
              "to": "/signup",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "className": "control",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="control"
              >
                Log In
              </button>,
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log In",
                "className": "control",
              },
              "ref": null,
              "rendered": "Log In",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
